src/components/kubectl/logs.ts

```diff
@@ -0,0 +1,161 @@
+    kubectl: Kubectl,
+    explorerNode: ResourceNode | undefined,
+    displayMode: LogsDisplayMode
+) {
+    if (explorerNode) {
+        return await getLogsForExplorerPod(kubectl, explorerNode, displayMode);
+    }
+
+    return logsForPod(kubectl, displayMode);
+}
+
+/**
+ * Fetch logs from a Pod, when selected from the Explorer.
+ */
+async function getLogsForExplorerPod(
+    kubectl: Kubectl,
+    explorerNode: ResourceNode,
+    displayMode: LogsDisplayMode
+) {
+    const namespace = explorerNode.namespace;
+    const podSummary = { name: explorerNode.id, namespace: namespace };
+    const container = await selectContainerForPod(podSummary);
+
+    // No container. How'd we get here. How does this Pod even work. What is going on.
+    if (!container) {
+        return;
+    }
+
+    getLogsForContainer(
+        kubectl,
+        podSummary,
+        container.name,
+        displayMode
+    );
+
+    return;
+}
+
+/**
+ * Fetches logs for a pod. If there are more than one containers,
+ * prompts the user for which container to fetch logs for.
+ */
+async function getLogsForPod(kubectl: Kubectl, podSummary: PodSummary, displayMode: LogsDisplayMode) {
+    if (!podSummary) {
+        vscode.window.showErrorMessage('Can\'t find a pod!');
+        return;
+    }
+
+    const container = await selectContainerForPod(podSummary);
+    if (!container) {
+        return;
+    }
+
+    getLogsForContainer(kubectl, podSummary, container.name, displayMode);
+}
+
+/**
+ * Gets the logs for a container in a provided pod, in a provided namespace, in a provided container.
+ */
+function getLogsForContainer(
+    kubectl: Kubectl,
+    podSummary: PodSummary,
+    containerName: string | undefined,
+    displayMode: LogsDisplayMode
+) {
+    let cmd = `logs ${podSummary.name}`;
+
+    if (podSummary.namespace) {
+        cmd = `${cmd} --namespace=${podSummary.namespace}`;
+    }
+
+    if (containerName) {
+        cmd = `${cmd} --container=${containerName}`;
+    }
+
+    if (displayMode === LogsDisplayMode.Follow) {
+        cmd = `${cmd} -f`;
+        kubectl.invokeInNewTerminal(cmd, `${podSummary.name}-${containerName}`);
+        return;
+    }
+
+    kubectl.invokeInSharedTerminal(cmd);
+}
+
+/**
+ * Searches for a pod yaml spec from the open document
+ * or from the currently selected namespace.
+ */
+async function logsForPod(kubectl: Kubectl, displayMode: LogsDisplayMode): Promise<void> {
+    const editor = vscode.window.activeTextEditor;
+
+    if (editor) {
+        return await logsForPodFromOpenDocument(kubectl, editor, displayMode);
+    }
+
+    return await logsForPodFromCurrentNamespace(kubectl, displayMode);
+}
+
+/**
+ * Finds a Pod from the open editor.
+ */
+async function logsForPodFromOpenDocument(kubectl: Kubectl, editor: vscode.TextEditor, displayMode: LogsDisplayMode) {
+    const text = editor.document.getText();
+    try {
+        const obj: {} = yaml.safeLoad(text);
+        if (!isPod(obj)) {
+            return;
```

Comment:
I think you have changed the logic here.  Previously if the open document was not a pod, it would fall through to the prompt case (same as if there was no active document).  It looks like now if the open document is not a pod it will silently do nothing.  The old behaviour was intentional and we should keep it.

Reply:
fixed!